### ISSUE TITLE
RFE: ensime-typecheck-subproject

### ISSUE THREAD

**Issue by peterbecich** [ensime/ensime-server#867]
Working through the "Functional Programming in Scala" project:  https://github.com/fpinscala/fpinscala

This project has code duplicated between subprojects, but no duplicated code within any single subproject.  `ensime-typecheck-all` seems to not keep these subprojects separate, so shows a lot of `defined twice` errors.

Is there a different function to typecheck just the 'working' subproject?

I think the `.ensime` file is fine.

```
:subprojects ((
  :name "answers"
  :module-name "answers"
  .
  .
  .
  :name "exercises"
  :module-name "exercises"
  .
  .
  )
```
ensime version 20150302.2023 and Emacs version 24.4.1

**Comment by edani** [comment]
At the moment, support for subprojects is very basic, ENSIME just merges all the paths together. See #716 .

**Comment by fommil** [comment]
Interesting. This particular use feels like a corner case... bordering on bad design by the upstream author. If you want to workaround it, I recommend creating two `.ensime` files: one for questions and one for answers.

**Comment by fommil** [comment]
actually, this would probably be solved by #839

**Comment by fommil** [comment]
implemented in ensime-sbt a few months ago, or perhaps still open via #993
